### PR TITLE
refactor(avoidance): separate shift line generate process

### PR DESCRIPTION
**Pull request by satoshi-ota**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at c6629fc</samp>

This pull request refactors the `AvoidanceModule` class to use a new class `ShiftLineGenerator` for handling the shift lines. The `ShiftLineGenerator` class is declared in a new header file `shift_line_generator.hpp` and defined in a new source file `shift_line_generator.cpp`. The pull request also moves a utility function `isComfortable` from the `AvoidanceModule` class to the `helper.hpp` file. The purpose of these changes is to simplify the code and avoid duplication and manual clearing of the shift lines.

## Tests performed

- [x] [PASS TIER IV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/97df888a-cb6a-5097-af94-44457f110591?project_id=prd_jt) ALL PASS

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/include/behavior_path_planner/utils/avoidance/avoidance_module_data.hpp [hunk 1]
```diff
@@ -495,8 +495,8 @@ struct AvoidancePlanningData
   // nearest object that should be avoid
   boost::optional<ObjectData> stop_target_object{boost::none};
 
-  // raw shift point
-  AvoidLineArray raw_shift_line{};
+  // // raw shift point
+  // AvoidLineArray raw_shift_line{};
```
Comment (rej55) [inline planning/behavior_path_planner/include/behavior_path_planner/utils/avoidance/avoidance_module_data.hpp:499]: Can we remove this?

**Review by rej55 — APPROVED** [review]

File: planning/behavior_path_planner/include/behavior_path_planner/utils/avoidance/avoidance_module_data.hpp (hunk 1 above)
Comment (satoshi-ota) [reply planning/behavior_path_planner/include/behavior_path_planner/utils/avoidance/avoidance_module_data.hpp:499]: @rej55 Ah.. Sorry, I'll remove it.

Comment (satoshi-ota) [reply planning/behavior_path_planner/include/behavior_path_planner/utils/avoidance/avoidance_module_data.hpp]: @rej55 Fixed.

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/avoidance/avoidance_module.hpp [hunk 2]
```diff
@@ -221,6 +224,11 @@ class AvoidanceModule : public SceneModuleInterface
    */
   void insertStopPoint(const bool use_constraints_for_decel, ShiftedPath & shifted_path) const;
 
+  /**
+   * @brief insert stop point to return original lane.
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_planner/include/behavior_path_planner/scene_module/avoidance/avoidance_module.hpp:228]:
```suggestion
   * @brief insert stop point in return path to original lane.
```

File: planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp [hunk 3]
```diff
@@ -496,24 +447,13 @@ void AvoidanceModule::fillShiftLine(AvoidancePlanningData & data, DebugData & de
...
-  const auto outlines = generateAvoidOutline(data, debug);
-
-  /**
-   * STEP2: Create rough shift lines.
-   */
-  data.raw_shift_line = applyPreProcess(outlines, debug);
-
-  /**
-   * STEP3: Create candidate shift lines.
+   * STEP1: Create candidate shift lines.
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp:450]: easy to understand whole flow!

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM!

**Pull request merged**